newrelic/resource_newrelic_alert_policy_channel.go

```diff
@@ -83,56 +99,103 @@ func resourceNewRelicAlertPolicyChannelRead(d *schema.ResourceData, meta interfa
 	}
 
 	d.Set("policy_id", policyID)
-	d.Set("channel_id", channelID)
+
+	_, channelIDOk := d.GetOk("channel_id")
```

Comment:
Can some of the `Set`s here be refactored to a `flatten` func?

Reply:
Updated